scripts/sct_label_vertebrae.py

```diff
@@ -59,116 +61,162 @@ def get_parser():
+        required=True,
+        help="Segmentation of the spinal cord. Example: t2_seg.nii.gz"
+    )
+    mandatory.add_argument(
+        '-c',
+        choices=['t1', 't2'],
+        required=True,
+        help="Type of image contrast. 't2': cord dark / CSF bright. 't1': cord bright / CSF dark"
+    )
+
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit."
+    )
+    optional.add_argument(
+        '-t',
+        metavar=Metavar.folder,
+        default=os.path.join(sct.__data_dir__, "PAM50"),
+        help="Path to template."
+    )
+    optional.add_argument(
+        '-initz',
+        metavar=Metavar.list,
+        type=list_type(',', int),
+        help=("R|Initialize using slice number and disc value. Example: 68,4 (slice 68 corresponds to disc C3/C4). "
+              "Example: 125,3\n"
+              "WARNING: Slice number should correspond to superior-inferior direction (e.g. Z in RPI orientation, but "
+              "Y in LIP orientation).")
+    )
+    optional.add_argument(
+        '-initcenter',
+        metavar=Metavar.int,
+        type=int,
+        help=("Initialize using disc value centered in the rostro-caudal direction. If the spine is curved, then "
+              "consider the disc that projects onto the cord at the center of the z-FOV.")
+    )
+    optional.add_argument(
+        '-initfile',
+        metavar=Metavar.file,
+        help="Initialize labeling by providing a text file which includes either -initz or -initcenter flag."
+    )
+    optional.add_argument(
+        '-initlabel',
+        metavar=Metavar.file,
+        help=("Initialize vertebral labeling by providing a nifti file that has a single disc label. An example of "
+              "such file is a single voxel with value '3', which would be located at the posterior tip of C2-C3 disc. "
+              "Such label file can be created using: sct_label_utils -i IMAGE_REF -create-viewer 3 ; or by using the "
+              "Python module 'detect_c2c3' implemented in 'spinalcordtoolbox/vertebrae/detect_c2c3.py'.")
+    )
+    optional.add_argument(
+        '-discfile',
+        metavar=Metavar.file,
+        help=("File with disc labels, which will be used to transform the input segmentation into a vertebral level "
+              "file. In that case, there will be no disc detection. The convention for disc labels is the following: "
+              "value=3 -> disc C2/C3, value=4 -> disc C3/C4, etc.")
+    )
+    optional.add_argument(
+        '-ofolder',
+        metavar=Metavar.file,
+        action=ActionCreateFolder,
+        default='',
+        help=("Output folder.")
+    )
+    optional.add_argument(
+        '-denoise',
+        choices=['0', '1'],
+        default='0',
+        help="Apply denoising filter to the data. Sometimes denoising is too aggressive, so use with care."
```

Comment:
What denoising filter does this use @jcohenadad?

Reply:
Looks like this flag is used in a call to `sct_maths`. The description there provides some extra contex, I thinkt:

https://github.com/neuropoly/spinalcordtoolbox/blob/004b3262d6cde2209d6af8c96f7e03370ada0928/scripts/sct_maths.py#L185-L192

Reply:
good point-- we should probably mention it:
~~~
help="Apply denoising filter (non-local means adaptative denoising) to the data. Sometimes denoising is too aggressive, so use with care."
~~~